include/seqan/align/dp_formula.h

```diff
@@ -110,101 +114,104 @@ typedef Tag<RecursionDirectionZero_> RecursionDirectionZero;
+template <typename TScoreNavigator,
+          typename TSequenceHValue,
+          typename TSequenceVValue,
+          typename TScoringScheme,
+          typename TRecursionDirection,
+          typename TDPProfile>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<ExtractedScoreValueType_<TScoreNavigator>>>,
+                            typename TraceBitMap_<ExtractedScoreValueType_<TScoreNavigator>>::Type)
+_computeScore(TScoreNavigator && dpNavi,
               TSequenceHValue const & seqHVal,
               TSequenceVValue const & seqVVal,
               TScoringScheme const & scoringScheme,
               TRecursionDirection const & recDir,
               TDPProfile const & dpProfile)
 {
-    typedef typename TraceBitMap_<TScoreValue>::Type TTraceValue;
+    using TScoreValue = ExtractedScoreValueType_<TScoreNavigator>;
 
-    TTraceValue traceDir = _doComputeScore(activeCell, previousDiagonal, previousHorizontal, previousVertical, seqHVal,
-                                           seqVVal, scoringScheme, recDir, dpProfile);
+    auto traceDir = _doComputeScore(std::get<0>(dpNavi), std::get<1>(dpNavi), std::get<2>(dpNavi),
+                                    std::get<3>(dpNavi), seqHVal, seqVVal, scoringScheme, recDir, dpProfile);
     if (IsLocalAlignment_<TDPProfile>::VALUE)
     {
-        TScoreValue cmp = cmpGt(createVector<TScoreValue>(1), activeCell._score);
-        _setScoreOfCell(activeCell, TraceBitMap_<TScoreValue>::NONE, cmp);
-        _setHorizontalScoreOfCell(activeCell, TraceBitMap_<TScoreValue>::NONE, cmp);
-        _setVerticalScoreOfCell(activeCell, TraceBitMap_<TScoreValue>::NONE, cmp);
+        TScoreValue cmp = cmpGt(createVector<TScoreValue>(1), _scoreOfCell(std::get<0>(dpNavi)));
+        _setScoreOfCell(std::get<0>(dpNavi), TraceBitMap_<TScoreValue>::NONE, cmp);
         return blend(traceDir, TraceBitMap_<TScoreValue>::NONE, cmp);
     }
-
     return traceDir;
 }
 
 // ----------------------------------------------------------------------------
 // Function _doComputeScore                        [RecursionDirectionDiagonal]
 // ----------------------------------------------------------------------------
 
-template <typename TScoreValue, typename TGapCosts, typename TSequenceHValue, typename TSequenceVValue, typename TScoringScheme,
+template <typename TScoreValue, typename TGapCosts,
+          typename TSequenceHValue,
+          typename TSequenceVValue,
+          typename TScoringScheme,
           typename TDPProfile>
-inline typename TraceBitMap_<TScoreValue>::Type
+inline auto
 _doComputeScore(DPCell_<TScoreValue, TGapCosts> & activeCell,
-                DPCell_<TScoreValue, TGapCosts> const & previousDiagonal,
-                DPCell_<TScoreValue, TGapCosts> const & /*previousHorizontal*/,
-                DPCell_<TScoreValue, TGapCosts> const & /*previousVertical*/,
+                DPCell_<TScoreValue, TGapCosts> const & prevDiag,
```

Comment:
I changed the interface quite often to evaluate different possibilities. 
I will change these back.